src/main/java/io/supertokens/cronjobs/bulkimport/ProcessBulkImportUsers.java

```diff
@@ -0,0 +1,353 @@
+        BulkImportUserUtils bulkImportUserUtils = new BulkImportUserUtils(allUserRoles);
+
+        for (BulkImportUser user : users) {
+            processUser(app, user, bulkImportUserUtils, bulkImportSQLStorage);
+        }
+    }
+
+    @Override
+    public int getIntervalTimeSeconds() {
+        if (Main.isTesting) {
+            Integer interval = CronTaskTest.getInstance(main).getIntervalInSeconds(RESOURCE_KEY);
+            if (interval != null) {
+                return interval;
+            }
+        }
+        return BulkImport.PROCESS_USERS_INTERVAL_SECONDS;
+    }
+
+    @Override
+    public int getInitialWaitTimeSeconds() {
+        if (Main.isTesting) {
+            Integer waitTime = CronTaskTest.getInstance(main).getInitialWaitTimeInSeconds(RESOURCE_KEY);
+            if (waitTime != null) {
+                return waitTime;
+            }
+        }
+        return 0;
+    }
+
+    private synchronized Storage getBulkImportProxyStorage(TenantIdentifier tenantIdentifier)
+            throws InvalidConfigException, IOException, TenantOrAppNotFoundException, DbInitException,
+            StorageQueryException {
+        String userPoolId = StorageLayer.getStorage(tenantIdentifier, main).getUserPoolId();
+        if (userPoolToStorageMap.containsKey(userPoolId)) {
+            return userPoolToStorageMap.get(userPoolId);
+        }
+
+        TenantConfig[] allTenants = Multitenancy.getAllTenants(main);
+
+        Map<ResourceDistributor.KeyClass, JsonObject> normalisedConfigs = Config.getNormalisedConfigsForAllTenants(
+                allTenants,
+                Config.getBaseConfigAsJsonObject(main));
+
+        for (ResourceDistributor.KeyClass key : normalisedConfigs.keySet()) {
+            if (key.getTenantIdentifier().equals(tenantIdentifier)) {
+                SQLStorage bulkImportProxyStorage = (SQLStorage) StorageLayer.getNewBulkImportProxyStorageInstance(main,
+                        normalisedConfigs.get(key), tenantIdentifier, true);
+
+                userPoolToStorageMap.put(userPoolId, bulkImportProxyStorage);
+                bulkImportProxyStorage.initStorage(false, new ArrayList<>());
+                // `BulkImportProxyStorage` uses `BulkImportProxyConnection`, which overrides the `.commit()` method on the Connection object.
+                // The `initStorage()` method runs `select * from table_name limit 1` queries to check if the tables exist but these queries
+                // don't get committed due to the overridden `.commit()`, so we need to manually commit the transaction to remove any locks on the tables.
+
+                // Without this commit, a call to `select * from bulk_import_users limit 1` in `doesTableExist()` locks the `bulk_import_users` table,
+                // causing other queries to stall indefinitely.
+                bulkImportProxyStorage.commitTransactionForBulkImportProxyStorage();
```

Comment:
can this be moved to the storage layer? it can be override the initStorage function, call the super impl, and then call commitTransactionForBulkImportProxyStorage?

Reply:
done.